File: chapter-1/problems/java/prime_factorization.java
import java.util.*;

public class PrimeFactorization{
    static Scanner scanner = new Scanner(System.in);

    static void primeFactorization(int n, List<Integer> factors){
        if(n<=1){
            return;
        }
        for(int i = 2;i*i<=n;i++){
            while(n%i==0){
                factors.add(i);
                n/=i;
            }
        }

        if(n>1){
            factors.add(n);
        }
    }
    

    public static void main(String args[]){
        List<Integer> factors = new ArrayList<>();
        System.out.print("Enter the value of n : ");
        int n = scanner.nextInt();
        primeFactorization(n,factors);
        System.out.println(factors.toString());
    }
}

File: chapter-1/problems/cpp/prime_factorization.cpp
#include<bits/stdc++.h>

using namespace std;

void prime_factorization(int n, unordered_set<int>& factors) {
    for (int i = 2; i * i <= n; i++) {
        while (n % i == 0) {
            factors.insert(i);
            n /= i;
        }
    }
    if (n > 1) factors.insert(n);
}

int main(){
    int n;
    cout<< "Enter the value of n : ";
    cin>>n;
    unordered_set<int> factors;

    prime_factorization(n,factors);

    for(auto i: factors){
        cout<<i<<" ";
    }
    return 0;
}